docs/release.md

```diff
@@ -0,0 +1,78 @@
+
+- T-30 (owned by Dev): Cut release branch.
+- From T-29 to T-16 (owned by QA, assisted by Dev): Testing period.
+    - Run a performance comparison with bounded and unbounded tests, Postgres and MySQL, using the latest stable release of Mattermost as both the base and new builds. The test will pass if and only if:
+        - All four tests finish successfully.
+        - There is no significant difference between the base and new builds (they're the same build, and thus should ideally output the same exact results).
+    - If bugs are found:
+        - Fix them in master.
+        - Cherry-pick them to the release branch.
+        - Run the test again.
+- T-16 end of day (owned by Dev): judge whether the tests successfully passed and decide if the release will happen the next day.
+- T-15 (owned by Dev): If the tests passed, release a new version. If not, skip the release for this month.
+
+### Owners
+
+There are two roles specified in this process:
+
+- Dev: the main owner of the whole process, in charge of cutting the release branch, assisting during the testing period, judging whether the release is ready and releasing the new version.
+- QA: the main owner of the testing period, in charge of running the tests and filing the bugs.
+
+The current people in charge of these roles are:
+- Dev: @agarciamontoro
+- QA: @DHaussermann
+
+In the case that the primary Dev and QA are not available, we define Dev and QA backups:
+- Dev backups: @streamer45, @agnivade, @isacikgoz
+- QA backups: @agarciamontoro, @streamer45, @agnivade, @isacikgoz
+
+
+## Release
+
+This section outlines the technicalities of the release itself, once the new version is tested and approved.
+
+We use a combination of `Makefile` and [Goreleser](https://goreleaser.com/) to automate this process, which consists of two steps:
+1. Prepare the release and get the automatically created PR merged.
+2. Do the actual release.
+
+### Prepare the release
+
+There are certain changes that we need to do for every release (for now, simply updating the default load-test URL in the config sample). You can automatically apply those changes with the following command:
+
+```sh
+make prepare-release NEXT_VER=v1.1.1
+```
+
+Follow the instructions in the output to create the PR. Once it's merged, continue with the next step.
+
+### Actual release
+
+With the PR already merged, you can now finish the process with the following command:
+
+```sh
+make release NEXT_VER=v1.1.1
+```
+
+`goreleaser` needs to use a Github token to create the release for you. You can define it inline with the `GITHUB_TOKEN` variable or write it to the default `~/.config/goreleaser/github_token` file. Read [`goreleaser` docs](https://goreleaser.com/scm/github/) for more info.
```

Comment:
This part is just moved from `developing.md`, it now makes more sense to have it here.